src/components/Notifications/NotificationsLog/NotificationsLogDateFilter.tsx

```diff
@@ -0,0 +1,295 @@
+}
+
+const CustomDateFilter: React.FunctionComponent<CustomDateFilterProps> = (
+  props
+) => {
+  const maxDate = React.useMemo(() => new Date(), []);
+  const minDate = React.useMemo(
+    () =>
+      add(maxDate, {
+        days: -14,
+      }),
+    [maxDate]
+  );
+
+  const startRangeValidators = React.useMemo(
+    () => [
+      (date) => {
+        if (isBefore(date, minDate)) {
+          return 'Date is before the retention policy';
+        } else if (isAfter(date, maxDate)) {
+          return 'Date is after today';
+        } else if (props.period[1] && isAfter(date, props.period[1])) {
+          return 'Start date must be before end date';
+        }
+
+        return '';
+      },
+    ],
+    [minDate, maxDate, props.period]
+  );
+
+  const endRangeValidators = React.useMemo(
+    () => [
+      (date) => {
+        if (isBefore(date, minDate)) {
+          return 'Date is before the retention policy';
+        } else if (isAfter(date, maxDate)) {
+          return 'Date is after today';
+        } else if (props.period[0] && isBefore(date, props.period[0])) {
+          return 'End date must be after start date';
+        }
+
+        return '';
+      },
+    ],
+    [minDate, maxDate, props.period]
+  );
+
+  const setStartDate = React.useCallback(
+    (start: string) => {
+      const setPeriod = props.setPeriod;
+      const startDate = parseISO(start);
+      setPeriod(
+        produce((draft) => {
+          draft[0] = startDate;
+          if (!draft[1]) {
+            draft[1] = min([add(startDate, { days: 1 }), maxDate]);
+          }
+        })
+      );
+    },
+    [props.setPeriod, maxDate]
+  );
+
+  const setEndDate = React.useCallback(
+    (end: string) => {
+      const setPeriod = props.setPeriod;
+      setPeriod(
+        produce((draft) => {
+          draft[1] = parseISO(end);
+        })
+      );
+    },
+    [props.setPeriod]
+  );
+
+  const startValue: string | undefined = React.useMemo(
+    () => (props.period[0] ? format(props.period[0], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+  const endValue: string | undefined = React.useMemo(
+    () => (props.period[1] ? format(props.period[1], 'yyyy-MM-dd') : undefined),
+    [props.period]
+  );
+
+  const startDateRef = useRef<DatePickerRef>(null);
+  const endDateRef = useRef<DatePickerRef>(null);
+
+  const onClickStartDateInput = React.useCallback(() => {
+    startDateRef.current?.setCalendarOpen(true);
+  }, [startDateRef]);
+
+  const onClickEndDateInput = React.useCallback(() => {
+    endDateRef.current?.setCalendarOpen(true);
+  }, [endDateRef]);
+
+  const startDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickStartDateInput,
+    }),
+    [onClickStartDateInput]
+  );
+
+  const endDateInputProps = React.useMemo<TextInputProps>(
+    () => ({
+      ...dateInputProps,
+      onClick: onClickEndDateInput,
+    }),
+    [onClickEndDateInput]
+  );
```

Comment:
I don't think we have to wrap these functions in memoes and callbacks. It's enough to pass objects to `DatePicker` component

For start datepicker:
```JSX
inputProps={{
  ...dateInputProps,
  onClick: () => startDateRef.current?.setCalendarOpen(true)
}}
```

For end datepicker:
```JSX
inputProps={{
  ...dateInputProps,
  onClick: () => endDateRef.current?.setCalendarOpen(true)
}}
```

Reply:
Fixed